docker/policyconfiguration/naming.go

```diff
@@ -43,14 +45,25 @@ func DockerReferenceNamespaces(ref reference.Named) []string {
 	// iteration matches the host name (for any namespace).
 	res := []string{}
 	name := ref.Name()
+	var subdomain bool
 	for {
-		res = append(res, name)
-
-		lastSlash := strings.LastIndex(name, "/")
-		if lastSlash == -1 {
-			break
+		if !subdomain {
+			res = append(res, name)
+			lastSlash := strings.LastIndex(name, "/")
+			if lastSlash == -1 {
+				s := "true"
+				subdomain, _ = strconv.ParseBool(s)
+				continue
+			}
+			name = name[:lastSlash]
+		} else {
+			res = append(res, name)
+			firstDot := strings.Index(name, ".")
+			if firstDot == -1 {
+				break
+			}
+			name = fmt.Sprint("*", name[firstDot-1:])
 		}
-		name = name[:lastSlash]
 	}
```

Comment:
Could https://golang.org/pkg/net/url/#Parse be helpful here?

Reply:
@ashcrow do you mean rewrite the entire splitting including namespaces/subdomains using url.Parse? Or just for stripping the port number? I used portNumColon for detecting port, but I guess I could use SplitHostPort.

Reply:
@ashcrow is this what you meant?

```
-       portNumColon := strings.Index(name, ":")
-       name = name[:portNumColon]
+       HostPort, err := url.Parse(name)
+       if err != nil {
+               panic(err)
+       }
+       name, _, _ = net.SplitHostPort(HostPort.Host)
```

Reply:
Probably not `url.Parse`, because the inputs are not URLs (e.g. no escaping, and not most of the semantics.)

`SplitHostPort` does a fair bit of extra work, which can be ignored for the input that is actually acceptable to `reference.DomainRegexp`, but it also fails on port-less host names; so it would not simplify much.

Reply:
@lsm5 yes, but it was more specifically a question as if it would help in this situation. I defer to @mtrmac's note.